### ISSUE TITLE
Support setting the color of the undercurl on terminals

### ISSUE THREAD

**Issue by RedBeard0531** [vim/vim#2918]
Related to https://github.com/vim/vim/issues/1306. When support for t_Cs and t_Ce was added for terminals it didn't get support for setting the color to guisp, so it just keeps the forground color. I've found that this seems to do the right thing for red, but that forces that color on all undercurls rather than letting SpellCap (also used for warnings) be blue.

```vim
    set t_Ce=[4:0m[59m
    set t_Cs=[4:3m[58;5;9m
    highlight SpellBad cterm=undercurl guisp=Red guibg=Black
    highlight SpellCap cterm=undercurl guisp=Blue guibg=Black
```

(Unfortunately github doesn't show the escape chars before the [s, and the `let &t_Cs='\e[...'` form doesn't seem to work for me)

**Comment by RedBeard0531** [comment]
Escape codes taken from https://github.com/kovidgoyal/kitty/blob/master/protocol-extensions.asciidoc#colored-and-styled-underlines

**Comment by k-takata** [comment]
(off topic)

> and the `let &t_Cs='\e[...'` form doesn't seem to work for me

You need to use double quotes instead of single quotes.

**Comment by joshklod** [comment]
Bump...

See also #819 #1773 #2405

**Comment by lacygoill** [comment]
Wasn't this fixed by [8.2.0863](https://github.com/vim/vim/releases/tag/v8.2.0863)?

**Comment by chrisbra** [comment]
closing then.
